### PR TITLE
send flow nits

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
- disable tab swipe inside stack nav pages
- confirm screen: reset on tapping amount
- fix search bar clicks on Home Screen (seem to be broken by swipe nav?)
- fix android tab swipe
- use react-native-screen for optimization

### CODE REVIEW

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx [hunk 1]
```diff
@@ -159,6 +159,7 @@ function HomeScreenInner({ account }: { account: Account }) {
         onScrollEndDrag={onScrollEndDrag}
         onScroll={scrollHandler}
         scrollEventThrottle={8}
+        keyboardShouldPersistTaps="handled"
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HomeScreen.tsx:162]: 👍

File: apps/daimo-mobile/src/view/screen/link/NoteScreen.tsx [hunk 2]
```diff
@@ -48,6 +48,9 @@ export default function NoteScreen(props: Props) {
 }
 
 function NoteScreenInner({ route, account }: Props & { account: Account }) {
+  const nav = useNav();
+  useDisableTabSwipe(nav);
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/link/NoteScreen.tsx:52]: nice hook

File: apps/daimo-mobile/src/view/shared/AmountInput.tsx [hunk 3]
```diff
@@ -119,7 +125,21 @@ function AmountInput({
...
+    innerRef?.current?.focus();
+    if (onFocus) onFocus();
+  }, [innerRef, onFocus]);
+
+  const nav = useNav();
+
+  useEffect(() => {
+    // Re-focus after screen transition animations finish.
+    // This is a workaround for a bug in react-navigation where autoFocus
+    // doesn't persist across screen animations.
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/shared/AmountInput.tsx:138]: 👀 
sg

File: apps/daimo-mobile/src/view/shared/nav.ts [hunk 4]
```diff
@@ -122,3 +122,13 @@ async function goTo(nav: MainNav, link: DaimoLink) {
 export function navResetToHome(nav: MainNav) {
   nav.reset({ routes: [{ name: "HomeTab" }] });
 }
+
+export function useDisableTabSwipe(nav: MainNav) {
+  useEffect(() => {
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/shared/nav.ts:127]: minor but i'd do something like

```ts
const p = nav.getParent()
if (p == null) return
p.setOptions({ swipeEnabled: false });
return () => p.setOptions({ swipeEnabled: true });
```

**Review by dcposch — APPROVED** [review]
Looks great to me

File: apps/daimo-mobile/src/view/shared/nav.ts (hunk 4 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/shared/nav.ts:128]: done

**Pull request merged**